dbms/src/Parsers/ExpressionListParsers.cpp

```diff
@@ -595,6 +595,69 @@ bool ParserNullityChecking::parseImpl(Pos & pos, ASTPtr & node, Expected & expec
     return true;
 }
 
+bool ParserDateOperatorExpression::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    auto begin = pos;
+
+    /// If no DATE keyword, go to the nested parser.
+    if (!ParserKeyword("DATE").ignore(pos, expected))
+        return next_parser.parse(pos, node, expected);
+
+    ASTPtr expr;
+    if (!ParserStringLiteral().parse(pos, expr, expected))
```

Comment:
But why it's just for literals?
Does SQL standard support `DATE x`?

Reply:
It sayes this:
```
F051-01 DATE data type (including support of DATE literal)
— Subclause 5.3, “<literal>”: The <date literal> form of <datetime literal>
— Subclause 6.1, “<data type>”: The DATE <datetime type>
— Subclause 6.31, “<datetime value expression>”: For values of type DATE

<datetime literal> ::= <date literal> | <time literal> | <timestamp literal>
<date literal> ::= DATE <date string>
<time literal> ::= TIME <time string>
<timestamp literal> ::= TIMESTAMP <timestamp string>
```
I haven't found any place where it would be applyed to non literal. In MySQL it's also possible to apply DATE operator for timestamp string to exptract date. In PG it's possible to cropp seconds in timestamp string. But for SQL compatibility we need TIME, 6-digits afrer dot TIMESTAMP and lots of copmarison, timezones and operation results rules.